casper/src/main/scala/coop/rchain/casper/Estimator.scala

```diff
@@ -123,11 +123,12 @@ final class Estimator[F[_]: Sync: Log: Metrics: Span](
   ): F[Map[BlockHash, Long]] = {
     def hashParents(hash: BlockHash, lastFinalizedBlockNumber: Long): F[List[BlockHash]] =
       for {
-        currentBlockNumber <- blockDag.lookup(hash).map(_.get.blockNum)
-        result <- if (currentBlockNumber < lastFinalizedBlockNumber)
-                   List.empty[BlockHash].pure[F]
-                 else
-                   blockDag.lookup(hash).map(_.get.parents)
+        block              <- blockDag.lookupUnsafe(hash)
+        currentBlockNumber = block.blockNum
+        result = if (currentBlockNumber <= lastFinalizedBlockNumber)
```

Comment:
I think this is a valid change and it should be present, but now it breaks Estimator, when LCA is invalid block. 

Because invalid block has no children because of this 
https://github.com/rchain/rchain/blob/0b160341a4826357e6521fed0371544ac3a5933d/block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockMetadataStore.scala#L67
As invalid block cannot be parents (it can only be justifications), it won't have children when added to DAG.

So, when doing ranking, this code 
https://github.com/rchain/rchain/blob/633cd59bc1256a0c1d18424eef0537be9b8796c3/casper/src/main/scala/coop/rchain/casper/Estimator.scala#L165-L182
exits straight away


Want to reiterate that having old invalid block from slashed validator is questionable by itself, but looks like for now we don't touch this part.

So might be reverting this change makes sense for now. And to make LFS work - we have to download parents of the lowest required blocks as well.

Reply:
Makes sense to download parents also for lowest of latest blocks.

I agree, invalid block in justification can be treated in a different way which can change which blocks need to be downloaded.